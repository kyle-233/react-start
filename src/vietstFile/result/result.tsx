type Student = {
  id?: number
  name?: string
}

interface ResultProps {
  students: Student[]
}

export const Result = ({ students }: ResultProps) => {
  return (
    <div>
      {!students.length ? (
        <h1>No Data</h1>
      ) : (
        students.map((student) => (
          <div key={student.id}>
            <div>
              <h1>{student.name}</h1>
            </div>
          </div>
        ))
      )}
    </div>
  )
}
