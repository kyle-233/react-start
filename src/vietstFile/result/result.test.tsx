import { render } from '@testing-library/react'
import { Result } from './result.tsx'

describe('Results', () => {
  test('should renders correctly with no stus', () => {
    const { asFragment } = render(<Result students={[]} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should renders correctly with some stus', () => {
    const stus = [
      {
        id: 1,
        name: 'Luna',
      },
    ]
    const { asFragment } = render(<Result students={stus} />)
    // 渲染快照结果是否与存档快照一致
    expect(asFragment()).toMatchSnapshot()
  })
})
