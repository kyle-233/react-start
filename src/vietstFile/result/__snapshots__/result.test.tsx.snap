// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Results > should renders correctly with no stus 1`] = `
<DocumentFragment>
  <div>
    <h1>
      No Data
    </h1>
  </div>
</DocumentFragment>
`;

exports[`Results > should renders correctly with some stus 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <h1>
          Luna
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
